main.go

```diff
@@ -168,10 +165,11 @@ func init() {
 	// enum flags
 	rootCmd.Flags().VarP(stageFlag, "stage", "s", fmt.Sprintf("set the stage to run (%s)", strings.Join(stageFlag.Allowed, ", ")))
 	rootCmd.Flags().VarP(loglevelFlag, "level", "l", fmt.Sprintf("set the log level (%s)", strings.Join(loglevelFlag.Allowed, ", ")))
-	rootCmd.Flags().VarP(ksProvider, "kubernetes-provider", "k", fmt.Sprintf("Kubernetes provider (%s)", strings.Join(ksProvider.Allowed, ", ")))
 	// rest of the flags
 	rootCmd.Flags().StringVarP(&config.DefaultConfig.Model, "model", "m", "generic", "model to build for, like generic or rpi4")
-	rootCmd.Flags().StringVar(&config.DefaultConfig.KubernetesVersion, "k8sversion", "latest", "Kubernetes version for provider")
+	rootCmd.Flags().StringVarP(&config.DefaultConfig.ProviderName, "provider", "k", "", fmt.Sprintf("Provider plugin"))
```

Comment:
ah I see it was renamed here, yup that's better, but then the k8sversion flag on the Dockerfile.test should be providerVersion